projects/plugins/jetpack/json-endpoints/jetpack/class.jetpack-json-api-plugins-modify-endpoint.php

```diff
@@ -107,24 +107,60 @@
 	)
 );
 
+/**
+ * Plugins modify endpoint class.
+ *
+ * POST  /sites/%s/plugins/%s
+ * POST  /sites/%s/plugins
+ */
 class Jetpack_JSON_API_Plugins_Modify_Endpoint extends Jetpack_JSON_API_Plugins_Endpoint {
-	// POST  /sites/%s/plugins/%s
-	// POST  /sites/%s/plugins
+
+	/**
+	 * The slug.
+	 *
+	 * @var string
+	 */
 	protected $slug = null;
+
+	/**
+	 * Needed capabilities.
+	 *
+	 * @var string
+	 */
 	protected $needed_capabilities = 'activate_plugins';
+
+	/**
+	 * Action.
+	 *
+	 * @var string
+	 */
 	protected $action = 'default_action';
+
+	/**
+	 * Expected actions.
+	 *
+	 * @var array
+	 */
 	protected $expected_actions = array( 'update', 'install', 'delete', 'update_translations' );
 
+	/**
+	 * Callback.
+	 *
+	 * @param string $path - the path.
+	 * @param int    $blog_id - the blog ID.
+	 * @param object $object - the object.
+	 */
 	public function callback( $path = '', $blog_id = 0, $object = null ) {
 		Jetpack_JSON_API_Endpoint::validate_input( $object );
 		switch ( $this->action ) {
 			case 'delete':
 				$this->needed_capabilities = 'delete_plugins';
+				// Allowing fallthrough.
```

Comment:
This seems more like a bug being flagged by phpcs than an intended fallthrough. As it is, the assignment here doesn't do anything since it gets overwritten on line 161.

Reply:
Think we can just remove line 157 altogether then? Or are we missing a `break` statement and this has just been bugged for a long time? I'm not sure how to test this to see what our expected behavior is.

Reply:
Good question. I think we're just missing the `break` statement and this has been bugged since 2016 and no one noticed, since WordPress does seem to have a 'delete_plugins' capability.

I'm fairly lost on how to test any of these. As far as I can tell, most of them get run by an xmlrpc backdoor rather than the normal site API.

Reply:
Added the break statement for now. We'll know where to look if things go sideways.